d2/src/main/java/com/linkedin/d2/balancer/simple/ClusterLoadBalancerSubscriber.java

```diff
@@ -68,4 +72,29 @@ protected void handleRemove(final String listenTo)
     _simpleLoadBalancerState.getClusterInfo().remove(listenTo);
     _simpleLoadBalancerState.notifyClusterListenersOnRemove(listenTo);
   }
+
+  /**
+   * Pick the active properties (stable or canary configs) based on canary distribution strategy.
+   * @param discoveryProperties a composite properties containing all data on the cluster store (stable configs, canary configs, etc.).
+   * @return the picked active properties
+   */
+  private ClusterProperties pickActiveProperties(final ClusterProperties discoveryProperties)
+  {
+    ClusterProperties pickedProperties = discoveryProperties;
+    CanaryDistributionProvider.Distribution distribution = CanaryDistributionProvider.Distribution.STABLE;
+    if (discoveryProperties instanceof ClusterStoreProperties) // this should always be true since the serializer returns the composite class
```

Comment:
Should we be doing something if the assumption isn't true? Logging at least?

Reply:
we have changed the serialiers to always return the composite class. If it doesn't, the unit tests will fail.

Reply:
Are they unit tests for this class? If not it'd be possible to make that change, see the tests failing, and then fix the tests. But this would still break.

Reply:
there is a TestClusterLoadBalancerSubscriber (also one for the service load balancer subscriber). If this breaks, that test will fail.